server/server.go

```diff
@@ -248,6 +256,42 @@ func (server *HTTPServer) voteOnRule(writer http.ResponseWriter, request *http.R
 	}
 }
 
+func (server *HTTPServer) validateClusterAndRuleID(
+	clusterID types.ClusterName, ruleID types.RuleID, writer http.ResponseWriter,
+) bool {
+	clusterExists, err := server.Storage.CheckIfClusterExists(clusterID)
+	if err != nil {
+		log.Err(err)
+		responses.Send(http.StatusInternalServerError, writer, err.Error())
+		return false
+	}
+	if !clusterExists {
+		responses.Send(
+			http.StatusNotFound,
+			writer,
+			(&storage.ItemNotFoundError{ItemID: clusterID}).Error(),
+		)
+		return false
+	}
```

Comment:
Why not return `storage.ItemNotFoundError` from storage and let `handleServerError` handle it? Or do you not like that approach?

Reply:
hm, maybe actually better, thanks

Reply:
where is this `handleServerError`?

Reply:
i guess you dont have master merged

Reply:
server/errors.go

Reply:
nevermind, I'm using an old version of base branch

Reply:
fixed